### PR TITLE
Optuna hpo

### PR DESCRIPTION
**Pull request by kimdanni**

### CODE REVIEW

File: waffle_hub/schema/enums.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
+from enum import Enum
+
+
+class HPOMethod(Enum):
```
Comment (dustjr1565) [inline waffle_hub/schema/enums.py:4]: It would be good to inherit from baseEnum in __init__.py.

File: tests/test_hpo.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        "lr0": [0.005, 0.05],
+        "lrf": [0.001, 0.005],
+        "mosaic": [0.6, 1],
+        "cos_lr": (True, False),
+        "hsv_h": [0.01, 0.02],
+        "hsv_s": [0.01, 0.02],
+        "hsv_v": [0.01, 0.02],
+        "translate": [0.09, 0.11],
+        "scale": [0.45, 0.55],
+        "mosaic": [0.6, 1],
```
Comment (dustjr1565) [inline tests/test_hpo.py:22]: There are two mosaic field.

File: waffle_hub/core/hpo/base_hpo.py [hunk 3]
```diff
@@ -0,0 +1,84 @@
+from abc import abstractmethod
+
+from waffle_hub.hub import Hub
+
+
+class BaseHPO:
```
Comment (dustjr1565) [inline waffle_hub/core/hpo/base_hpo.py:6]: Where is the class inherited from?

**Comment by kimdanni** [comment]
Add hpoconfig

**Comment by Jiyong-Jeon** [comment]
You need to modify ".github/workflows".

File: tests/tests.py [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+    assert "best_params" in result
+    assert "best_score" in result
+    assert Path(hub.root_dir / hub.name / "evaluate.json").exists()
+    assert Path(hub.root_dir / hub.name / "hpo.json").exists()
+    assert Path(hub.root_dir / hub.name / "metrics.json").exists()
+    assert Path(hub.root_dir / hub.name / "hpo" / last_trial_directory_name).exists()
+    assert Path(hub.root_dir / hub.name / db_name).exists()
+
+
+if __name__ == "__main__":
```
Comment (dustjr1565) [inline tests/tests.py:72]: why does it exist in test code??

File: waffle_hub/core/hpo/base_hpo.py [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+    def __init__(
+        self,
+        percent: float = 0.3,
```
Comment (dustjr1565) [inline waffle_hub/core/hpo/base_hpo.py:9]: The docstring and the param seem to differ

**Pull request closed without merging**